src/expr/tests/testdata/reduce

```diff
@@ -318,7 +318,7 @@ reduce
     (call_variadic coalesce
         [
             (call_variadic (record_create ["f1" "f2"]) [#0 #1])
-            (call_variadic (record_create ["f1" "f2"]) [null null])
+            (call_variadic (record_create ["f1" "f2"]) [(null int32) (null int32)])
```

Comment:
The verification that `MirScalarExpr` does not change the type of an expression detected that this test was accidentally written with inconsistent types.